eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqliteNetworkDb.scala

```diff
@@ -61,7 +60,17 @@ class SqliteNetworkDb(val sqlite: Connection) extends NetworkDb with Logging {
       case Some(v@1) =>
         logger.warn(s"migrating db $DB_NAME, found version=$v current=$CURRENT_VERSION")
         migration12(statement)
-      case Some(CURRENT_VERSION) => () // table is up-to-date, nothing to do
+      case Some(CURRENT_VERSION) =>
+        // We clean up channels that contain an invalid channel update (e.g. missing htlc_maximum_msat).
+        val invalidChannels = statement.executeQuery("SELECT short_channel_id, channel_update_1, channel_update_2 FROM channels").map(rs => {
+          val shortChannelId = rs.getLong("short_channel_id")
+          val validChannelUpdate1 = rs.getBitVectorOpt("channel_update_1").forall(channelUpdateCodec.decode(_).isSuccessful)
+          val validChannelUpdate2 = rs.getBitVectorOpt("channel_update_2").forall(channelUpdateCodec.decode(_).isSuccessful)
+          (shortChannelId, validChannelUpdate1 && validChannelUpdate2)
+        }).filter { case (_, isValid) => !isValid }.map { case (scid, _) => scid }.toSeq
```

Comment:
```suggestion
        }).collect { case (scid, false) => scid }.toSeq
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2379/commits/55b919f8ae079a559e7200c39b74f26bbc896304